05_speed_test_comparison_dplyr_duckdb.qmd

```diff
@@ -0,0 +1,299 @@
+---
+title: "DuckDB Evaluation"
+date: today
+format: html
+toc: true
+params:
+  n_trials: 5
+  run_loop: no
+editor_options: 
+  chunk_output_type: console
+---
+
+# Introductions
+
+The purpose of this project is to provide a high-level evaluation of [DuckDB](https://duckdb.org) as a solution for projects involving databases at BC Stats.
+
+# Advantages of DuckDB
+
+Section left blank for Jon.
+
+# Disadvantages of DuckDB
+
+Section left blank for Jon.
+
+# Loading Data into DuckDB
+
+Section left blank for Jon. Note you can use bash scripts here -- maybe useful for embedding the dbt code?
+
+```{bash}
+# echo hello jon!!!
+```
+
+# Speed Tests
+
+## Introduction
+
+```{r setup}
+pacman::p_load(tictoc, tidyverse, duckdb, zeallot, pryr, readr)
+
+csv_path = "G:/Operations/Data Science and Analytics/2024_bcstats_db/csvs"
+
+csv_names = c(
+  "98-401-X2021006_English_CSV_data_BritishColumbia-utf8.csv",
+  "BC Stat Population Estimates_20240429.csv",
+  "CLR_EXT_20231227.csv",
+  "bca_folio_gnrl_property_values_20240825_WEEKLY.csv",
+  "bca_folio_sales_20240825_WEEKLY.csv"
+)
+```
+
+In this section, we compare the speed of certain operations between `DuckDB`, `duckplyr` and `dplyr` in R. The operations are:
+
+-   loading .csv files as standalone R data frames or into a DuckDB database
+-   querying either R data frames using `dbplyr` or `duckplyr` syntax, or querying DuckDB databases.
```

Comment:
@jonjunduan it looks like the speed tests include both loading csv files and doing some basic analysis - so assuming you can get this to run, it's probably more than sufficient?